tests/integration/components/changeset-test.js

```diff
@@ -172,6 +172,75 @@ test('nested object updates when set without a validator', async function(assert
   assert.equal(find('h1').textContent.trim(), 'foo bar', 'should update observable value');
 });
 
+test('nested key error clears after entering valid input', async function(assert) {
```

Comment:
@krishandley Oh, one more thing – these two tests are nearly the same, I'd like to refactor this into table-driven form like in https://github.com/poteto/ember-changeset/pull/226#discussion_r153973472.

Lemme know if you have time to work on this, if not I'll take a look later today or tomorrow.

Reply:
Yea that’s fine, I can do that.

Reply:
A loop isn't going to work, because variables in the hbs helper don't work.

We can use Ember.HTMLBars.compile, but... we need to add
`app.import('vendor/ember/ember-template-compiler.js');` for 2.13 + and
`app.import('bower_components/ember/ember-template-compiler.js');` for below to the ember-cli-build.js file.

Both do work when running the tests for a single version, but I'm not sure how to target the ember version being used before importing. So I'm at a dead end trying to get that to work, and try/catch didn't work either.

So the simplest solution is just to have multiple inputs, and error divs in the same test/template. If you don't mind that instead?

Reply:
Hmm, maybe we could set a selector instead of the template string interpolation? For example:

```js
// this is the changeset
this.set('c', c);

// this is test data, where s === 'person.name.parts.first.validation'
this.set('selector', s);
```

```hbs
<input value={{get c selector}} onchange={{action (mut (get c selector))}}>
```

Not completely sure that'll work – would you mind giving it a shot?

Reply:
Good idea, worked like a charm 😄

Reply:
Spoke too soon, 1.13 needs doesn't have the get helper. And adding the get helper addon broke other tests :(

Reply:
:( which tests are failing?

Reply:
Well, the summary says 8 tests failed, but looking through the only test names which I can see in the failed output are the ones in the loop... But they all say this. Uncaught Error: Stream error: setValue not implemented.

Feel free to take a look at merging the tests yourself, I'll try again tomorrow with fresh eyes if don't get to it.

Reply:
Just took a look, yeah let's not go down this timesink. 😛

This is fine for now then, great job @krishandley!